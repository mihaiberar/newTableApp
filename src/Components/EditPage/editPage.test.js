import { EditPage } from './editPage.jsx'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import React from 'react'
import { Router } from 'react-router-dom'
import { Context } from '../Context/context'

// jest.mock('react-router-dom', () => ({
//   ...jest.requireActual('react-router-dom'),
//   useParams: jest.fn()
// }))

describe('componentToTest', () => {
  // beforeEach(() => {
  //   useParamsMock.mockImplementation(() => {
  //     return {
  //       id: ''
  //     }
  //   })
  //   useContextMock.mockImplementation(() => ['my data'])
  // })
  // afterEach(() => {
  //   jest.clearAllMocks()
  // })
  it('should capture Snapshot of <EditPage/> component', () => {
    // useContextMock.mockReturnValue('Test Value')

    const history = createMemoryHistory()
    const data = [{ name: 'Mihai' }]
    const { container } = render(
      <Router history={history}>
        <Context.Provider value={[data]}>
          <EditPage />
        </Context.Provider>
      </Router>
    )
    expect(container).toMatchSnapshot()
  })
})
