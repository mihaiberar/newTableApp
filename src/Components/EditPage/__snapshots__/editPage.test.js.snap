// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componentToTest should capture Snapshot of <EditPage/> component 1`] = `
<div>
  <div
    class="editPage"
  >
    <div
      class="editHeader"
    >
      <h1>
        Edit:
      </h1>
      <button
        style="width: 30px; height: 30px; margin-top: 30px; margin-left: 30px; background-color: rgb(241, 41, 41);"
      >
        âœŽ
      </button>
    </div>
    <div
      class="idName"
    >
      <input
        class="idInput"
        name="id"
        placeholder="id"
        readonly=""
        type="text"
        value=""
      />
      <input
        class="nameInput"
        name="name"
        placeholder="name"
        readonly=""
        style="outline: none;"
        type="text"
        value="Mihai"
      />
    </div>
    <div
      class="ownerUpdateAt"
    >
      <input
        class="ownerInput"
        name="owner"
        placeholder="owner"
        readonly=""
        style="outline: none;"
        type="text"
        value=""
      />
      <input
        class="updatedAtInput"
        name="updated at"
        placeholder="updated at"
        readonly=""
        style="outline: none;"
        type="text"
        value=""
      />
    </div>
    <div
      class="cancelSaveBtns"
    >
      <a
        href="/"
      >
        <button
          class="cancelBtn"
        >
          Cancel
        </button>
      </a>
      <a
        href="/"
      >
        <button
          class="saveBtn"
        >
          Save
        </button>
      </a>
    </div>
  </div>
</div>
`;
